x-pack/plugins/observability_solution/slo/public/components/slo/burn_rate/alert_details/components/custom_panels/custom_kql/log_rate_analysis_panel.tsx

```diff
@@ -0,0 +1,323 @@
+    to: timeRange.lte ? new Date(timeRange.lte) : new Date(),
+  };
+}
+
+interface Props {
+  slo: GetSLOResponse;
+  alert: BurnRateAlert;
+  rule: BurnRateRule;
+}
+
+interface SignificantFieldValue {
+  field: string;
+  value: string | number;
+  docCount: number;
+  pValue: number | null;
+}
+
+export function LogRateAnalysisPanel({ slo, alert, rule }: Props) {
+  const services = useKibana().services;
+  const { dataViews: dataViewsService, observabilityAIAssistant } = services;
+  const ObservabilityAIAssistantContextualInsight =
+    observabilityAIAssistant?.ObservabilityAIAssistantContextualInsight;
+  const [dataView, setDataView] = useState<DataView | undefined>();
+  const [esSearchQuery, setEsSearchQuery] = useState<QueryDslQueryContainer | undefined>();
+  const [logRateAnalysisParams, setLogRateAnalysisParams] = useState<
+    | { logRateAnalysisType: LogRateAnalysisType; significantFieldValues: SignificantFieldValue[] }
+    | undefined
+  >();
+  const params = slo.indicator.params as KQLCustomIndicator['params'];
+  const groupBy = slo.groupBy;
+  const groupings = slo.groupings;
+  const { index } = params;
+  const { data: dataViews = [] } = useFetchDataViews();
+
+  useEffect(() => {
+    const getDataView = async () => {
+      const getDataViewByIndexPattern = (indexPattern: string) =>
+        dataViews.find((dataView0) => dataView0.title === indexPattern);
+
+      const dataViewId = getDataViewByIndexPattern(index)?.id;
+      if (dataViewId) {
+        const sloDataView = await dataViewsService.get(dataViewId);
+        setDataView(sloDataView);
+        getQuery();
+      }
+    };
+
+    const getQuery = () => {
+      const esSearchRequest = getESQueryForLogRateAnalysis(
+        params,
+        groupBy,
+        groupings
+      ) as QueryDslQueryContainer;
+      if (esSearchRequest) {
+        setEsSearchQuery(esSearchRequest);
+      }
+    };
+    getDataView();
+  }, [index, dataViews, params, dataViewsService, groupBy, groupings]);
+
+  // Identify `intervalFactor` to adjust time ranges based on alert settings.
+  // The default time ranges for `initialAnalysisStart` are suitable for a `1m` lookback.
+  // If an alert would have a `5m` lookback, this would result in a factor of `5`.
+  // If an alert is just starting, the visible time range to look back might not cover the
+  // long window yet, only then we'll take the short window to look back,
+  // otherwise it will be the long window.
+  const alertActionGroup = getActionGroupFromReason(alert.reason);
+  const relatedWindow = (
+    (alert.fields[ALERT_RULE_PARAMETERS]?.windows ?? []) as WindowSchema[]
+  ).find((window: WindowSchema) => window.actionGroup === alertActionGroup);
+
+  const longWindowValue = relatedWindow?.longWindow.value;
+  const longWindowUnit = relatedWindow?.longWindow.unit;
+  const longWindowLookbackDuration =
+    longWindowValue && longWindowUnit
+      ? moment.duration(longWindowValue as number, longWindowUnit as any)
```

Comment:
can we use `DurationUnit` instead of `any`? Same for `shortWindowUnit` in `shortWindowLookbackDuration`.

Reply:
that should work I think, the casting to `number` and `any` is just a leftover from copy pasting from the threshold rule.

Reply:
I pushed a fix